### PR TITLE
Benchmarking

### PR DESCRIPTION
**Pull request by lalithsuresh**
Adds a runnable jar that benchmarks histograms. Also involves removing several null checks.

### CODE REVIEW

File: platform/src/main/java/org/hillview/table/api/IStringColumn.java [hunk 1]
```diff
@@ -18,19 +18,18 @@
...
 public interface IStringColumn extends IColumn {
     @Override
-    default double asDouble(final int rowIndex, @Nullable final IStringConverter converter) {
+    default double asDouble(final int rowIndex, final IStringConverter converter) {
         if (isMissing(rowIndex))
             throw new MissingException(this, rowIndex);
         final String tmp = this.getString(rowIndex);
         //noinspection ConstantConditions
-        return Converters.checkNull(converter).asDouble(tmp);
+        return converter.asDouble(tmp);
```
Comment (mbudiu-vmw) [inline platform/src/main/java/org/hillview/table/api/IStringColumn.java]: the comment above it probably unnecessary

File: platform/src/main/java/org/hillview/table/columns/BaseArrayColumn.java [hunk 2]
```diff
@@ -29,24 +27,24 @@
...
             throw new InvalidParameterException("Size must be positive: " + size);
-        if (this.description.allowMissing && !this.description.kind.isObject())
-            this.missing = new BitSet(size);
+        this.missing = new BitSet(size);
     }
 
     @Override
     public boolean isMissing(final int rowIndex) {
-        return this.description.allowMissing && Converters.checkNull(this.missing).get(rowIndex);
+        // Ignore the analysis complaint. this.missing can not be null.
```
Comment (mbudiu-vmw) [inline platform/src/main/java/org/hillview/table/columns/BaseArrayColumn.java]: Why is it complaining?
There is an official IDEA comment for this.
Same observation below

File: platform/src/main/java/org/hillview/table/NoOpStringConverter.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import org.hillview.table.api.IStringConverter;
+import org.hillview.table.api.IStringConverterDescription;
+
+import javax.annotation.Nullable;
+
+/**
+ * A string converter that does nothing
+ */
+public class NoOpStringConverter implements IStringConverter, IStringConverterDescription {
+    private static final NoOpStringConverter INSTANCE = new NoOpStringConverter();
```
Comment (mbudiu-vmw) [inline platform/src/main/java/org/hillview/table/NoOpStringConverter.java]: Why not make this public and remove getInstance()?

File: platform/src/main/java/org/hillview/table/NoOpStringConverter.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+import org.hillview.table.api.IStringConverter;
+import org.hillview.table.api.IStringConverterDescription;
+
+import javax.annotation.Nullable;
+
+/**
+ * A string converter that does nothing
+ */
+public class NoOpStringConverter implements IStringConverter, IStringConverterDescription {
```
Comment (mbudiu-vmw) [inline platform/src/main/java/org/hillview/table/NoOpStringConverter.java]: let's call this NoStringConverter.
It's not a no-op, it's a throw.

File: platform/src/main/java/org/hillview/table/columns/BaseArrayColumn.java (hunk 2 above)
Comment (lalithsuresh) [reply platform/src/main/java/org/hillview/table/columns/BaseArrayColumn.java]: Stale comment. Will remove.

File: platform/src/main/java/org/hillview/table/NoOpStringConverter.java (hunk 3 above)
Comment (lalithsuresh) [reply platform/src/main/java/org/hillview/table/NoOpStringConverter.java]: While not strictly necessary here, I did it out of habit (it's cleaner to access such singletons behind a method in the event we want to say, log every access to the API).

**Pull request merged**